src/App.js

```diff
@@ -12,11 +14,10 @@ function App() {
   return (
     <div className="App">
       <Header setUsername={setUsername}/>
+      <NavBar />
       <Routes>
-        <Route
-          path="/"
-          element={<ArticleList />}
-        ></Route>
+        <Route path="/" element={<ArticleList />}></Route>
+        <Route path="/articles/topics/:topic" element={<ArticlesByTopic/>}></Route>
```

Comment:
You should be able to reuse your articlesList component here - this is creating a lot of extra code, and you're repeating a large chunk of your code base.